### PR TITLE
Don't cache lastPath between import wizard runs

### PR DESCRIPTION
**Pull request by baroquebobcat**
- reset lastPath and settings on new import wizard start
- reorder path analysis to allow executable render case to work
- extract path analysis into helper
- use same path analysis for both validation and rendering

### CODE REVIEW

File: src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java [hunk 1]
```diff
@@ -112,65 +120,84 @@ protected boolean isExtraSettingModified() {
...
                 Messages.createMessageDialogRemover(getProject()).run();
               });
+            }
           }
-        }
-      });
+        });
+        break;
+      default:
+        throw new ConfigurationException("unexpected state: " + pathFileType);
```
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: The compile error suggests this function cannot throw `ConfigurationException`, so using `UIUtil.invokeLaterIfNeeded` like above might be more appropriate.

Comment (baroquebobcat) [reply src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: last minute adjustment fail 🤦‍♂️

Comment (baroquebobcat) [reply src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: Does `UIUtil.invokeLaterIfNeeded` allow the exception to bubble up somewhere visible?

Comment (wisechengyi) [reply src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: Yes the example above has `Pants Failure: ...`, which shows up as 
screen shot 2018-07-26 at 1 47 50 pm

You can test it by changing pants script to return false
```
$ cat pants
false
```

File: src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java [hunk 2]
```diff
@@ -112,65 +120,87 @@ protected boolean isExtraSettingModified() {
...
                 Messages.createMessageDialogRemover(getProject()).run();
               });
+            }
           }
-        }
-      });
+        });
+        break;
+      default:
+        UIUtil.invokeLaterIfNeeded((Runnable) () -> {
```
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: Hmm selecting a directory would trigger this..so maybe noop, which is the status quo?

Comment (baroquebobcat) [reply src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: hm. I don't think so. It should hit https://github.com/pantsbuild/intellij-pants-plugin/pull/362/files/655250383b1e924edb3c14e9d144a69aa9f1c5b6#diff-42c61d4fa92b800ccf4d203aca9fc5feR151

Comment (wisechengyi) [reply src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: screen shot 2018-07-26 at 3 10 55 pm

Comment (baroquebobcat) [reply src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: nice

**Pull request merged**